examples/error-handling/error_handling.bal

```diff
@@ -9,34 +31,56 @@ type AccountNotFoundErrorData record {
+const ACCOUNT_NOT_FOUND = "AccountNotFound";
+const INVALID_ACCOUNT_ID = "InvalidAccountID";
+type AccountNotFoundError error<ACCOUNT_NOT_FOUND | INVALID_ACCOUNT_ID, AccountNotFoundErrorData>;
 
 function getAccountBalance(int accountID) returns int|AccountNotFoundError {
-    // Returns an error if the `accountID` is less than zero.
+    // Values for the error detail mapping should be passed as named arguments.
     if (accountID < 0) {
-        string errorReason = "Account Not Found";
-        AccountNotFoundErrorData errorDetail = {
-            accountID: accountID
-        };
+        // Returns an error with "InvalidAccountID" as the reason if the `accountID` is less than zero.
         AccountNotFoundError accountNotFoundError =
-                                            error(errorReason, errorDetail);
+                                            error(INVALID_ACCOUNT_ID, accountID = accountID);
         return accountNotFoundError;
+    } else if (accountID > 100) {
+        // Returns an error with "AccountNotFound" as the reason if the `accountID` above hundred.
+        AccountNotFoundError accountNotFoundError =
+                                            error(ACCOUNT_NOT_FOUND, accountID = accountID);
     }
-    // Returns a value if the `accountID` is greater than zero.
+    // Returns a value if the `accountID` is in between zero and hundred inclusive.
     return 600;
 }
 
+
+
+
 public function main() {
-    var result = getAccountBalance(-1);
-    // If the `result` is an `int`, then print the value.
+    // Define an error of the generic error type.
+    error simpleError = error("SimpleErrorType", message = "Simple error occured");
+    // Print the error reason and the `message` field from the error detail.
+    // `message` is an optional field in the generic error `Detail` record.
+    io:println("Error: ", simpleError.reason(),
+                ", Message: ", simpleError.detail()?.message);
+
+    var result = getTypeId("Joined");
     if (result is int) {
-        io:println("Account Balance: ", result);
-    // If an error is returned, print the reason and the account ID from the detail map.
-    // The `.reason()` and `.detail()` built-in methods can be called on variables of
-    // the type `error` to retrieve the reason and details of the error.
+        io:println("Account type ID: ", result);
     } else {
         io:println("Error: ", result.reason(),
-                   ", Account ID: ", result.detail().accountID);
+                   ", Account type: ", result.detail().accountType);
+    }
+
+    var result2 = getAccountBalance(-1);
+    // If the `result` is an `int`, then print the value.
+    if (result2 is int) {
+        io:println("Account Balance: ", result2);
+        // If an error is returned, print the reason and the account ID from the detail record.
+        // The `.reason()` and `.detail()` built-in methods can be called on variables of
```

Comment:
Move these comments to L62 - where we are using the methods for the first time in this BBE